### ISSUE TITLE
Can't fetch data: No Data / 400 Error PageSpeed insights

### ISSUE THREAD

**Issue by thedavemurphy** [google/site-kit-wp#243]
#59  is showing that Analytics, Page Speed and Search console are connected but no date is being fetched. No error message

#150

![download (1)](https://user-images.githubusercontent.com/41326532/61295438-ee530d80-a7cf-11e9-8530-338b5d1c077b.png)
![download (2)](https://user-images.githubusercontent.com/41326532/61295458-f90da280-a7cf-11e9-92eb-659f273787a2.png)

- [Site Health Info](https://user-images.githubusercontent.com/24247981/64017784-e9f67000-cb43-11e9-9bde-dff3d1f0447c.png)
- [XHR response](https://user-images.githubusercontent.com/24247981/64017869-1a3e0e80-cb44-11e9-81a6-d8f0a692b665.png)

Same issue under following conditions
- All plugins disabled
- Modsecurity disabled
- Clear cache

## Screenshots

#150

## Additional Context

 - PHP Version: 
 - Plugin Version [e.g. 22]

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

## Changelog entry

*

**Comment by jamesozzie** [comment]
@thedavemurphy Thanks for reporting. In order to troubleshoot further can provide the following:
1. From your Site Kit dashboard can you share a screenshot of what you see
2. Can you share any [console errors](https://i.imgur.com/f5WfLb3.png) you see from within your Site Kit dashboard.

**Comment by iamshaharyar** [comment]
@jamesozzie I am facing the same issue. I have tried resetting the site kit and also disabled all plugins to check but it's still not working. 
1. [Site Kit dashboard](http://prntscr.com/offkrr) 
2. [Console errors](http://prntscr.com/offqnt)

**Comment by jamesozzie** [comment]
@iamshaharyar @thedavemurphy Are you using a security or firewall extension or software? This could be interfering with your Site Kit reports. 

If you are using iThemes security can you uncheck [this](https://i.imgur.com/7OyyYyI.png) option from the general settings and see if it resolves the issue. If you are not using ithemes can you let me know of any security plugins or software you are using, including any additional firewall settings or security steps in place with your web host, such as Modsecurity or any cloudflare configurations.

**Comment by iamshaharyar** [comment]
@jamesozzie I am using Wordfence in my website and also using Cloudflare as well. Can you kindly let me know which settings in need to alter to make it work?

**Comment by jamesozzie** [comment]
@iamshaharyar I have default settings enable for both and I don't see any immediate options in either that might be causing this issue. Do you have any security or firewall enabled at your web hosting level, such as modsecurity?

**Comment by iamshaharyar** [comment]
@jamesozzie I do have ModSecurity enabled at my web hosting level so I contacted them and turned it off for this website but still everything is just same and not working.

**Comment by jamesozzie** [comment]
@iamshaharyar Can you try and clear your cache now, and temporarily disable any security or firewall plugins before trying again?

**Comment by iamshaharyar** [comment]
@jamesozzie I have tried that all but nothing changed.

**Comment by iamshaharyar** [comment]
@jamesozzie Still no solution found for this issue?

**Comment by jamesozzie** [comment]
@iamshaharyar I know you have already attempted many of these steps but can you try the following, in order.
- Reset Site Kit
- Clear your browser cache & delete cookies
- Deactivate all plugins (including Cloudflare, ithemes security, WordFence etc) other than Site Kit 
- Change to an alternate WordPress theme (to rule out any interfering functions/features)
- Activate Site Kit using an incognito window 

If the above fails please let me know of the following so we can investigate further:
- PHP / Web Host Environment
- Security / Firewall plugins in use
- Console Errors (if different to what was provided previously)

**Comment by jamesozzie** [comment]
@iamshaharyar There is an open issue (#339) which will hopefully resolve the problem you are encountering with the failing request

**Comment by iamshaharyar** [comment]
Hey @jamesozzie, on issue #339 they only mentioned how to reproduce that error and doesn't said anything about how to resolve this issue because I have tried almost everything but still have that blank page where there is no data.

**Comment by jamesozzie** [comment]
@iamshaharyar If you can share the following we might be able to notice something:
- Your [Site Health info](https://i.imgur.com/Rtkmsba.png) 
- Any [XHR response](https://i.imgur.com/jLBwBdH.png) on the non responsive dashboard

**Comment by iamshaharyar** [comment]
@jamesozzie it doesn't look secure to share all my "Site Health info" here!

**Comment by iamshaharyar** [comment]
Here are the required details @jamesozzie,

- [Site Health Info](https://user-images.githubusercontent.com/24247981/64017784-e9f67000-cb43-11e9-9bde-dff3d1f0447c.png)
- [XHR response](https://user-images.githubusercontent.com/24247981/64017869-1a3e0e80-cb44-11e9-81a6-d8f0a692b665.png)

**Comment by jamesozzie** [comment]
@iamshaharyar Thanks for the update, very useful. Can you also share the header information from your XHR response, on the top 2 resources?

**Comment by iamshaharyar** [comment]
@jamesozzie you're always welcome! Is that the [header information](https://user-images.githubusercontent.com/24247981/64025683-ace7a900-cb56-11e9-8ca1-481266b2a582.png) we are looking for?

**Comment by jamesozzie** [comment]
@iamshaharyar Thats it, thanks for that. We'll investigate from our side and escalate if required or revert back.

**Comment by aaemnnosttv** [comment]
The `rest_missing_callback_param` error you're getting is from WP core because the parameter is either missing or `null`. It appears that the parameter is there as it is the first item in the query string. It could be that something is breaking the JSON which is preventing it from being parsed, which also results in a `null` value.

Would you please copy the request headers like so and paste here? (not as an image)

![image](https://user-images.githubusercontent.com/1621608/64027584-abfc4a80-cb49-11e9-8e8c-683a8805671f.png)

Thanks!

**Comment by iamshaharyar** [comment]
@aaemnnosttv sure I can do that but I am not getting any option of "Copy request headers" on my side. These are the only [options](https://user-images.githubusercontent.com/24247981/64028430-174f1800-cb5c-11e9-81f5-99f0aaaf83c6.png) I am getting.

**Comment by aaemnnosttv** [comment]
In that case, please choose this option and share the resulting file:

![image](https://user-images.githubusercontent.com/1621608/64029323-02b75380-cb4d-11e9-958f-0b8ec2858b6c.png)

**Comment by iamshaharyar** [comment]
@aaemnnosttv here is the [HAR](https://www.dropbox.com/s/9smuth0r5m0iw9h/www.wpglue.com.har?dl=0) file.

**Comment by aaemnnosttv** [comment]
@iamshaharyar please email me at evan.mattson at 10up.com for more information.

**Comment by iamshaharyar** [comment]
@jamesozzie @aaemnnosttv Did we found what's wrong?

**Comment by aaemnnosttv** [comment]
Hi @iamshaharyar I didn't find anything yet, thanks for following up. The request sent appears to be correct, it seems as if something is interfering with it before it reaches your server perhaps. Do you use any kind of web application firewall (e.g. Cloudflare or Sucuri)?

**Comment by iamshaharyar** [comment]
@aaemnnosttv oh, I am actually using Cloudflare services. Do I need to make any modifications there?

**Comment by aaemnnosttv** [comment]
@iamshaharyar I'm not sure but it would be worth disabling anything other than DNS there temporarily to rule that out as a problem. I'm specifically thinking about the WAF they offer but it's probably easier to click the orange cloud for your domain to put it into dns-only mode for testing.

**Comment by iamshaharyar** [comment]
@aaemnnosttv I temporarily paused all cloudflare services except DNS and then reset all Site Kit configurations and created a new project again but still nothing get resolved and its exactly like before!

**Comment by aaemnnosttv** [comment]
@iamshaharyar I ran into a similar problem in another area recently and I think it is failing due to a subsequent internal request which is being made - not the main one we see in the console there. I'm going to look into this a bit more using the data you provided to see if we can locate where the error is coming from.

**Comment by iamshaharyar** [comment]
@aaemnnosttv that would be great if we can find out why we are facing this issue. I am always available to provide any kind of data which would be helpful to resolve it.

**Comment by RazaGR** [comment]
any update on this issue?

**Comment by RazaGR** [comment]
I found a solution, I was using Permalink settings to display URL with Post Name, when i set it to basic, there was no console error displayed and I was able to see the data, now to make the pretty URL work again with Post Name, I had to add this in my nginx config

```
location ~ ^/wp-json/ {
   rewrite ^/wp-json/(.*?)$ /?rest_route=/$1 last;
}
```

**Comment by iamshaharyar** [comment]
@RazaGR adding that URL rewrite rule in your .htaccess file resolved the issue?

**Comment by RazaGR** [comment]
@iamshaharyar I am using Nginx so above rule resolved it for me

**Comment by aaemnnosttv** [comment]
@RazaGR Just to clarify, what was your previous permalink structure? It sounds like you were using `/%postname%/` correct?  

The WP REST API uses `/wp-json/` as the prefix when "pretty permalinks" are enabled but we don't assume this, and the proper request URL format is handled by `wp.apiFetch`. You shouldn't have to set this manually at the server level.

**Comment by RazaGR** [comment]
@aaemnnosttv /%postname%/ correct? Yes, 

> You shouldn't have to set this manually at the server level.

I had to otherwise the request to `/wp-json/ `was failing with 400

**Comment by aaemnnosttv** [comment]
@RazaGR Unless you recently changed your permalink structure (new site perhaps?) and the rewrites weren't regenerated this shouldn't happen. Our E2E test suites would fail if this were the case as we use the same permalink structure for these.

I'm glad you were able to get it to work though. @iamshaharyar is this something you're able to test on your end?

**Comment by iamshaharyar** [comment]
@aaemnnosttv Sure I can try that on my end too. Just need to add this code to my .htaccess file right?

`location ~ ^/wp-json/ {
   rewrite ^/wp-json/(.*?)$ /?rest_route=/$1 last;
}`

**Comment by RazaGR** [comment]
@aaemnnosttv yes, I was testing on a new site, let me know if you need any more info

@iamshaharyar that rule is for nginx, for `.htaccess` file it should be 

```

RewriteEngine On
RewriteBase /
RewriteRule ^wp-json/(.*) /?rest_route=/$1 [L]

```

**Comment by jamesozzie** [comment]
@iamshaharyar Did you have any luck getting Site Kit setup in the end, by adding to your `.htaccess` file?

**Comment by aaemnnosttv** [comment]
@iamshaharyar this should be fixed as of [v1.1.2](https://github.com/google/site-kit-wp/releases/tag/1.1.2), can you confirm?
